### PR TITLE
Added section for watch.ocaml.org

### PR DESCRIPTION
**Pull request by guptadiksha307**
1) Wrote some useful bindings to the peertube REST API
2) Implemented all the various parts into ood (types, ocaml generation functions etc.)
3) Integrated the bindings into the ood cli commands
Final Output:

Screenshot 2021-07-23 at 5 08 55 PM

### CODE REVIEW

**Comment by tmattio** [comment]
Thanks, @guptadiksha307 - could you reopen the PR in #55 and push the code there instead, this will allow Patrick and I to follow the reviews, what's been addressed and what still needs changes

**Comment by patricoferris** [comment]
@tmattio we had all sorts of git problems I think, so just opening a new PR was easiest (I suggested it 😁 )

Edit: I'm also happy to re-review, we went over some of the code on a call :))

**Comment by patricoferris** [comment]
The CI error is because you forget to add `curly` as a dependency to the `ood-gen.opam` file (https://github.com/ocaml/ood/blob/main/ood-gen.opam). I talked a while a go about this, and what might be better is to convert your `curly` calls to `piaf` (another HTTP client library https://github.com/anmonteiro/piaf) that @tmattio already uses for the web scraping (e.g. https://github.com/ocaml/ood/blob/main/src/ood-gen/lib/news.ml#L32). I don't think it would be good to have two HTTP client dependencies :)) Does that make sense ?

**Comment by guptadiksha307** [comment]
yes @patricoferris I'll replace the `curly module`  with `piaf`.

File: src/ood-gen/bin/watch_scrape.ml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      else
+        let message = Status.to_string response.status in
+        Lwt.return (Error (`Msg message)) )
+
+let () =
+  let data () =
+    match get_sync "https://watch.ocaml.org/api/v1/videos" with
+    | Ok body -> Ezjsonm.value_from_string body
+    | Error error ->
+        let message = Error.to_string error in
```
Comment (patricoferris) [inline src/ood-gen/bin/watch_scrape.ml:68]: Why not `failwith message` -- this will fix the CI because right now this `message` is going unused.

**Comment by patricoferris** [comment]
Awesome work @guptadiksha307 -- as discussed offline we'll hook up the scraping to add it into `watch.yml` once we've finalised what videos are going into the peertube instance so the current state is perfect, thanks for all the hardwork this is a great addition to ood :))

**Comment by avsm** [comment]
Congratulations, this is fantastic work @guptadiksha307!

**Pull request merged**